### DISCUSSION TITLE
[gatsby-plugin-schema-snapshot] schema.gql output does not allow for developers to track and merge changes

### DISCUSSION THREAD

**Discussion by jordan-caldwell** [gatsbyjs/gatsby#32412] [Ideas / Feature Requests]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

The schema.gql appears to get completely rewritten each time the schema changes. This makes managing this file in source control difficult, especially when other developers need to see what was changed. 

It would be ideal if there was some mechanism for sorting the file in order for the output to be more consistent. 

### Reproduction Link

https://cant-share-the-code-because-its-private.com

### Steps to Reproduce

1. Have an existing schema.gql file in source control.
2. Add a new node type
3. Rebuild schema.gql

### Expected Result

The only changes to the existing schema.gql file should be a new timestamp and the new type for the merge. 

### Actual Result

The schema.gql file gets entirely rewritten with git being unable to track changes.

### Environment

```shell
System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 91.0.4472.114
    Firefox: 88.0.1
    Safari: 14.1
  npmPackages:
    gatsby: ^3.5.0 => 3.8.0
    gatsby-plugin-postcss: ^4.6.0 => 4.8.0
    gatsby-plugin-react-helmet: ^4.5.0 => 4.8.0
    gatsby-plugin-react-svg: ^3.0.1 => 3.0.1
    gatsby-plugin-sass: ^4.8.0 => 4.8.0
    gatsby-plugin-schema-snapshot: ^2.8.0 => 2.8.0
    gatsby-source-drupal: ^4.9.0 => 4.9.0
```

**Comment by vladar** [comment]
The way I understand what you are saying is that the output of `gatsby-plugin-schema-snapshot` is non-deterministic, so types and fields are printed in a different order when some change occurs and you just want to make sure that types/fields are sorted/deterministic in the output?

Is it correct to put it this way?

If yes, then it makes perfect sense to me. It should be relatively straightforward to implement. We print types here:

https://github.com/gatsbyjs/gatsby/blob/5f406a9848a1eb2f9e0462cd51d6f544e244bb03/packages/gatsby/src/schema/print.js#L152

Essentially we need to sort `typeDefs` by type name before printing.

Happy to accept a PR with this enhancement if you are interested!

**Reply by baparham** [reply]
are you still open to PRs? this still seems to be an issue
